HaulageApplication/HaulageApp/ViewModels/LoginViewModel.cs

```
@@ -0,0 +1,41 @@
+using CommunityToolkit.Mvvm.Input;
+using CommunityToolkit.Mvvm.ComponentModel;
+using HaulageApp.Models;
+using HaulageApp.Data;
+using Microsoft.Extensions.Logging;
+
+namespace HaulageApp.ViewModels;
+
+public partial class LoginViewModel : ObservableObject
+{
+    private readonly HaulageDbContext _context;
+    
+    public LoginViewModel(HaulageDbContext dbContext)
+    { 
+        _context = dbContext;
+    }
+    
+    public string Email { get; set; }
+    public string Password { get; set; }
+    
+    [RelayCommand]
+    private async Task Login()
+    {
+        if (IsCredentialCorrect(Email, Password))
```

Add try-catch block to catch database connection errors

---

Also, should we be handling cases if Email and Password are `null`. We should throw an error there?

---

I think a better way to handle nulls is to set any blank inputs to "", then you don't need to actually handle them. IT will be handled as an incorrect attempt.